Source/engine/random.hpp

```diff
@@ -72,6 +72,15 @@ uint32_t GenerateSeed();
  */
 int32_t GenerateRnd(int32_t v);
 
+/**
+ * @brief Generates a random value in [a, b] range using the vanilla RNG
+ * @param a - start of the range
+ * @param b - end of the range
+ * @param shift - if values should be multiplied by 64
+ * @return A random value in [a, b] range
+ */
+int32_t RndBetween(int32_t a, int32_t b, bool shift = false);
```

Comment:
In #6079 I provided a param to switch between half-open (`[a, b)`) and closed (`[a, b]`) ranges, if you wanted to expand the use to the other areas which use a half-open range something like
```suggestion
enum class RndRange {
  HalfOpen,
  Closed
};
int32_t RndBetween(int32_t a, int32_t b, RndRange mode = RndRange::Closed, bool shift = false) {
  // shift stuff omitted
  return a + GenerateRnd(b - a + (mode == RndRange::Closed ? 1 : 0));
}
```
would give that flexibility. An example of where this would be used in when generating gold values in items.cpp:GetItemAttrs().

Reply:
...I just realised a precursor to that PR was merged and `RandomIntBetween(min, max, isHalfOpen)` already exists in random.hpp. Apparently I added the function, used it once, and never finished the cleanup 🙃 

The version of `RndBetween()` you've added here is still useful for cases like setting item charges or AC (since we have to keep the biased behaviour to generate the same items as vanilla). The damage rolls identified in this PR would be more consistent with the bias-corrected version.